task/08-objects-tasks.js

```diff
@@ -111,34 +111,117 @@ function fromJSON(proto, json) {
+        return selector1.combine(combinator, selector2);
     },
 };
 
+class CssSelector {
+    constructor() {
+        this.levels = new Array(6).fill(false);
+        this.follow = [];
+        this.content = {
+            element: undefined,
+            id: undefined,
+            classes: [],
+            attributes: [],
+            pseudoClasses: [],
+            pseudoElement: undefined
+        };
+    }
+    checkLevel(level) {
+        let current = this.levels.slice(level + 1);
+        let isAdded = (isAdded) => isAdded;
+        let isOrderRight = current.some(isAdded);
+        if (isOrderRight) {
+            throw new Error('Selector parts should be arranged in the following order: element, id, class, attribute, pseudo-class, pseudo-element');
+        }
+        this.levels[level] = true;
+    }
+
+    element(value) {
+        this.checkLevel(0);
+        if (this.content.element === undefined) {
+            this.content.element = value;
+            return this;
+        }
+        throw new Error('Element, id and pseudo-element should not occur more then one time inside the selector');
+    }
+
+    id(value) {
+        this.checkLevel(1);
+        if (this.content.id === undefined) {
+            this.content.id = value;
+            return this;
+        }
+        throw new Error('Element, id and pseudo-element should not occur more then one time inside the selector');
+    }
+
+    class(value) {
+        this.checkLevel(2);
+        this.content.classes.push(value);
+        return this;
+    }
+
+    attr(value) {
+        this.checkLevel(3);
+        this.content.attributes.push(value);
+        return this;
+    }
+
+    pseudoClass(value) {
+        this.checkLevel(4);
+        this.content.pseudoClasses.push(value);
+        return this;
+    }
+
+    pseudoElement(value) {
+        this.checkLevel(5);
+        if (this.content.pseudoElement === undefined) {
+            this.content.pseudoElement = value;
+            return this;
+        }
+        throw new Error('Element, id and pseudo-element should not occur more then one time inside the selector');
+    }
+
+    combine(combinator, combinable) {
+        this.follow.push({ combinator: combinator, element: combinable });
+        return this;
+    }
+
+    stringify() {
+        return (this.content.element !== undefined ? this.content.element : '') +
+            (this.content.id !== undefined ? '#' + this.content.id : '') +
+            (this.content.classes.length ? '.' + this.content.classes.join('.') : '') +
+            (this.content.attributes.length ? this.content.attributes.map(elem => `[${elem}]`).join('') : '') +
+            (this.content.pseudoClasses.length ? ':' + this.content.pseudoClasses.join(':') : '') +
+            (this.content.pseudoElement !== undefined ? '::' + this.content.pseudoElement : '') +
+            (this.follow.length ? this.follow.map(elem => ` ${elem.combinator} ` + elem.element.stringify()).join('') : '');
```

Comment:
Please, try to rework all this block with such style:
`return (this.content.element || '') +
      (this.content.id  && `#${this.content.id}` || '') +`